### PR TITLE
Remove GPT-2  tokenizer dependancy from Deberta Tokenizers

### PR DESCRIPTION
**Pull request by RamitPahwa**
# What does this PR do?
Hi @sgugger 
Related to #19303 , 
- the GPT2Tokenizer dependency has been removed from DebertaTokenizer
- the GPT2TokenizerFast dependency has been removed from DebertaTokenizerFast

I ran` pytest tests/models/deberta/test_tokenization_deberta.py` which passed 
Thanks for reviewing!

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by RamitPahwa** [comment]
Hi @sgugger 
Please review the above PR when you get a moment.

File: src/transformers/models/deberta/tokenization_deberta.py [hunk 1]
```diff
@@ -63,6 +69,307 @@
...
+    pairs = set()
+    prev_char = word[0]
+    for char in word[1:]:
+        pairs.add((prev_char, char))
+        prev_char = char
+    return pairs
+
+
+# Copied from transformers.models.gpt2.tokenization_gpt2.GPT2Tokenizer
+class GPT2Tokenizer(PreTrainedTokenizer):
```
Comment (sgugger) [inline src/transformers/models/deberta/tokenization_deberta.py:113]: The goal is to copy and adapt. This should become `DebertaTokenizer :-)
```suggestion
# Copied from transformers.models.gpt2.tokenization_gpt2.GPT2Tokenizer with GPT2->Deberta,GPT-2->DeBERTa,"gpt2"->"microsoft/deberta-base"
class DebertaTokenizer(PreTrainedTokenizer):
```

File: src/transformers/models/deberta/tokenization_deberta_fast.py [hunk 2]
```diff
@@ -64,6 +70,130 @@
 }
 
 
+# Copied from transformers.models.gpt2.tokenization_gpt2_fast.GPT2TokenizerFast
+class GPT2TokenizerFast(PreTrainedTokenizerFast):
```
Comment (sgugger) [inline src/transformers/models/deberta/tokenization_deberta_fast.py:74]: Same here!
```suggestion
# Copied from transformers.models.gpt2.tokenization_gpt2_fast.GPT2TokenizerFast with GPT2->Deberta,GPT-2->DeBERTa,"gpt2"->"microsoft/deberta-base"
class GPT2TokenizerFast(PreTrainedTokenizerFast):
```

File: src/transformers/models/deberta/tokenization_deberta.py [hunk 3]
```diff
@@ -63,6 +69,307 @@
...
+    def _build_conversation_input_ids(self, conversation: "Conversation") -> List[int]:
+        input_ids = []
+        for is_user, text in conversation.iter_texts():
+            input_ids.extend(self.encode(text, add_special_tokens=False) + [self.eos_token_id])
+        if len(input_ids) > self.model_max_length:
+            input_ids = input_ids[-self.model_max_length :]
+        return input_ids
+
+
 class DebertaTokenizer(GPT2Tokenizer):
```
Comment (sgugger) [inline src/transformers/models/deberta/tokenization_deberta.py:373]: This should then be deleted.

File: src/transformers/models/deberta/tokenization_deberta_fast.py [hunk 4]
```diff
@@ -64,6 +70,130 @@
...
+        input_ids = []
+        for is_user, text in conversation.iter_texts():
+            input_ids.extend(self.encode(text, add_special_tokens=False) + [self.eos_token_id])
+
+        if len(input_ids) > self.model_max_length:
+            input_ids = input_ids[-self.model_max_length :]
+        return input_ids
+
+
 class DebertaTokenizerFast(GPT2TokenizerFast):
```
Comment (sgugger) [inline src/transformers/models/deberta/tokenization_deberta_fast.py:197]: This should then be deleted.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! The goal is to copy the GPT2 tokenizer objects while adapting them to the new DeBERTa model, not have a new GPT2Tokenzier in that file. I left comments on how to do this.

Note that you may not be able to copy the whole class since DeBERTa has some custom init, in this case just put the copied from on all methods.

File: src/transformers/models/deberta/tokenization_deberta.py (hunk 1 above)
Comment (RamitPahwa) [reply src/transformers/models/deberta/tokenization_deberta.py:113]: I see what you mean, i will raise a clean PR #19551

File: src/transformers/models/deberta/tokenization_deberta.py (hunk 3 above)
Comment (RamitPahwa) [reply src/transformers/models/deberta/tokenization_deberta.py:373]: I see what you mean, i will raise a clean PR https://github.com/huggingface/transformers/pull/19551

File: src/transformers/models/deberta/tokenization_deberta_fast.py (hunk 4 above)
Comment (RamitPahwa) [reply src/transformers/models/deberta/tokenization_deberta_fast.py:197]: I see what you mean, i will raise a clean PR https://github.com/huggingface/transformers/pull/19551

**Comment by RamitPahwa** [comment]
I raised a Clean PR #19551 for this work will close once that is merged, @sgugger Please review

**Pull request closed without merging**